thoth/user_api/api_v1.py

```diff
@@ -239,9 +239,23 @@ def get_analyze_status(analysis_id: str) -> typing.Tuple[typing.Dict[str, typing
     return _get_status("extract-packages", analysis_id, namespace=Configuration.THOTH_MIDDLETIER_NAMESPACE)
 
 
-def post_provenance_python(application_stack: dict, origin: str = None, debug: bool = False, force: bool = False):
+def post_provenance_python(
+    application_stack: dict,
+    debug: bool = False,
+    force: bool = False,
+    # Must be set to set protected fields
+    internal_secret: typing.Optional[str] = None,
+    # Protected fields
+    origin: str = None,
+):
     """Check provenance for the given application stack."""
+    protected_fields = ["origin"]
     parameters = locals()
+
+    if Configuration.INTERNAL_SECRET not in [internal_secret, None]:
+        for k in protected_fields:
+            parameters[k] = None
```

Comment:
It might be good to reject request rather than silently adjusting inputs - might be good for debugging and knowing the request as set was accepted by the system.